NebulaHost/PacketProcessors/Routers/StarBroadcastProcessor.cs

```diff
@@ -6,23 +6,23 @@
 namespace NebulaHost.PacketProcessors.Routers
 {
     [RegisterPacketProcessor]
-    class PlanetBroadcastProcessor : IPacketProcessor<PlanetBroadcastPacket>
+    class StarBroadcastProcessor : IPacketProcessor<StarBroadcastPacket>
     {
         private PlayerManager playerManager;
-        public PlanetBroadcastProcessor()
+        public StarBroadcastProcessor()
         {
             playerManager = MultiplayerHostSession.Instance.PlayerManager;
         }
-        public void ProcessPacket(PlanetBroadcastPacket packet, NebulaConnection conn)
+        public void ProcessPacket(StarBroadcastPacket packet, NebulaConnection conn)
         {
             Player player = playerManager.GetPlayer(conn);
             if (player != null)
             {
                 //Forward packet to other users
-                playerManager.SendRawPacketToPlanet(packet.PacketObject, packet.PlanetId, conn);
+                playerManager.SendRawPacketToStar(packet.PacketObject, packet.StarId, conn);
 
                 //Check if host is also target for the packet
-                if (GameMain.data.mainPlayer.planetId == packet.PlanetId)
+                if (GameMain.data.localStar?.id == packet.StarId)
                 {
                     MultiplayerHostSession.Instance.PacketProcessor.EnqueuePacketForProcessing(packet.PacketObject, conn);
                 }
```

Comment:
This is part of the `routing` packet. If client wants to broadcast something to the other users in the same star system, it will call:
`LocalPlayer.SendPacketToLocalStar( new MyPacket() )`
This method will take the `MyPacket` and it will put it inside of `StarBroadcastPacket` and sends it to the host.
The host will open the `StarBroadcastPacket`, takes the `MyPacket` that is inside, and sends it to other players. And this condition here is for checking if the host is also the intended recipient of that packet and if yes, it will take the `MyPacket` and puts it inside his queue. 

And basically, you can put any serializable packet inside the `StarBroadcastPacket`. It is just an envelope for the packet that will say who is the intended recipient if the packet that is inside.

Reply:
Yeah that part I understood, but the host should always be an intended recipient, since the host should have the entire game state synchronized at all time. Otherwise, if the host is in star system A and client in star system B, when the client performed an action, the host will never update his game state internally with the change. That means that if you then save the game on the host and reload the game later, all the changes done in other star systems will be lost.

Reply:
Aaaa right, I forgot about that :D :D